mk/Common.mk

```diff
@@ -231,123 +261,157 @@ include $(CFG_FLAGS_MAKE)
-COREV_SW_TOOLCHAIN  ?= /opt/corev
-COREV_VENDOR        ?= corev
-COREV_MARCH         ?= $(call RESOLVE_FLAG2,$(CFG_COREV_MARCH),rv32imc)
-COREV_CC            ?= gcc
-PULP_SW_TOOLCHAIN   ?= /opt/pulp
-PULP_VENDOR         ?= unknown
-PULP_MARCH          ?= $(call RESOLVE_FLAG2,$(CFG_PULP_MARCH),rv32imcxpulpv2)
-PULP_CC             ?= gcc
-LLVM_SW_TOOLCHAIN   ?= /opt/clang
-LLVM_VENDOR         ?= unknown
-LLVM_MARCH          ?= $(call RESOLVE_FLAG2,$(CFG_LLVM_MARCH),rv32imc)
-LLVM_CC             ?= cc
-
-CV_SW_TOOLCHAIN     ?= /opt/riscv
-CV_SW_VENDOR        ?= unknown
-CV_SW_MARCH         ?= $(call RESOLVE_FLAG2,$(CFG_CV_SW_MARCH),rv32imc)
-
-GNU_YES          = $(call IS_YES,$(GNU))
-PULP_YES         = $(call IS_YES,$(PULP))
-COREV_YES        = $(call IS_YES,$(COREV))
-LLVM_YES         = $(call IS_YES,$(LLVM))
-
-ifeq ($(shell $(CORE_V_VERIF)/mk/toolchain_check.sh $(GNU_YES) $(PULP_YES) $(COREV_YES) $(LLVM_YES)),1)
-$(error Multiple toolchains are enabled: GNU=${GNU_YES} PULP=${PULP_YES} COREV=${COREV_YES} LLVM=${LLVM_YES})
-endif
-
-RISCV             = $(CV_SW_TOOLCHAIN)
-RISCV_PREFIX      = riscv32-$(CV_SW_VENDOR)-elf-
-RISCV_EXE_PREFIX  = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC          = gcc
-RISCV_MARCH       = $(call RESOLVE_FLAG2,$(TEST_RISCV_MARCH),$(CV_SW_MARCH))
-RISCV_CFLAGS      = $(TEST_RISCV_CFLAGS)
-
-ifeq ($(GNU_YES),YES)
-ifeq ($(call IS_YES,$(TEST_GNU_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the GNU toolchain)
-endif
-RISCV            = $(GNU_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(GNU_VENDOR)-elf-
+# The so-called "COREV Environment Variables".
+#
+# First, check that the COREV environment variables have been defined.
+ifndef CV_SW_TOOLCHAIN
+$(error Must define CV_SW_TOOLCHAIN)
+endif
+ifndef CV_SW_PREFIX
+$(error Must define CV_SW_PREFIX)
+endif
+ifndef CV_SW_MARCH
+$(error Must define CV_SW_MARCH)
+endif
+#ifndef CV_SW_CFLAGS
+#$(warning CV_SW_CFLAGS is optional)
+#endif
+
+# Use these to set defaults for toolchain vars. A few typical examples...
+#RISCV            = /opt/riscv
+#RISCV_PREFIX     = riscv32-unknown-elf-
+#RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
+#RISCV_CFLAGS     =
+RISCV            = $(CV_SW_TOOLCHAIN)
+RISCV_PREFIX     = $(CV_SW_PREFIX)
 RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(GNU_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_GNU_MARCH),$(GNU_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_GNU_CFLAGS),$(GNU_CFLAGS))
-endif
 
-ifeq ($(COREV_YES),YES)
-ifeq ($(call IS_YES,$(TEST_COREV_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the COREV toolchain)
-endif
-RISCV            = $(COREV_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(COREV_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(COREV_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_COREV_MARCH),$(COREV_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_COREV_CFLAGS),$(COREV_CFLAGS))
-endif
+RISCV_CC         = gcc
```

Comment:
`gcc` is hard-coded here. But I think LLVM uses `cc`.
Maybe there should be a `CV_SW_CC`, while we still let RISCV_CC default to gcc?

Reply:
Sigh, you are right.  Thanks for pointing that out.